### PR TITLE
Theedge 3709 add activation key

### PR DESCRIPTION
**Pull request by acosferreira**
# Description

Include the select of activationKey to image create and update wizard.
Image Creation
image

Image Detail
image

Image Update
image

Fixes # (THEEDGE-3709)

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I run `npm run lint:js:fix` to check that my code is properly formatted

### CODE REVIEW

File: src/Routes/ImageManager/steps/ActivationKeys.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+          </Text>
+          <Text>
+            If you prefer to register later, review the instructions for manual
+            registration with remote host configuration.
+          </Text>
+          <Button
+            component="a"
+            target="_blank"
+            variant="link"
+            // icon={<ExternalLinkAltIcon />}
```
Comment (matoval) [inline src/Routes/ImageManager/steps/ActivationKeys.js:49]: There's some comments here I can probably be removed

File: src/Routes/ImageManagerDetail/ImageDetailTab.js [hunk 2]
```diff
@@ -170,6 +172,10 @@ const ImageDetailTab = ({
           <TextList component={TextListVariants.dl}>
             {buildTextList(userInfoMapper) || createSkeleton(2)}
           </TextList>
+          {/* <Text component={TextVariants.h2}>Activation Key </Text> */}
```
Comment (matoval) [inline src/Routes/ImageManagerDetail/ImageDetailTab.js:175]: Another comment

File: src/components/ImageCreator.js [hunk 3]
```diff
@@ -62,6 +63,10 @@ const CreateImageWizard = ({
         'ssh-input-field': {
           component: SSHInputField,
         },
+        'activation-keys-selector': {
+          // component: ActivationKeys,
```
Comment (matoval) [inline src/components/ImageCreator.js:67]: comment

File: src/components/form/ActivationKeysSelector.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    return (
+      <Button
+        component="a"
+        target="_blank"
+        variant="link"
+        icon={<ExternalLinkAltIcon />}
+        iconPosition="right"
+        isInline
+        href={
+          // isProd()
```
Comment (matoval) [inline src/components/form/ActivationKeysSelector.js:73]: comment

**Review by matoval — COMMENTED** [review]
LGTM, there's just a few comments that were left in

**Review by matoval — APPROVED** [review]

File: src/store/actions.js [hunk 5]
```diff
@@ -234,3 +236,19 @@ export const deleteEntity = (systems, displayName) => ({
     systems,
   },
 });
+
+export const loadActivationKey = (dispatch, pagination) => {
```
Comment (ldjebran) [inline src/store/actions.js:240]: where we are using this function ?

File: src/api/images/index.js [hunk 6]
```diff
@@ -23,6 +24,12 @@ export const fetchActiveImages = ({ limit = 100, offset = 0 } = {}) => {
   );
 };
 
+export const fetchActivationKeys = ({ limit = 10, offset = 0 } = {}) => {
```
Comment (ldjebran) [inline src/api/images/index.js:27]: think this limit number is too small, maybe 100

File: src/components/form/ActivationKeysSelector.js [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+  const [activationKeyList, setActivationKeyList] = useState(null);
+  const [isOpen, setIsOpen] = useState(false);
+  const [selected, setSelected] = useState(key);
+  const [isUpdate, setIsUpdate] = useState(false);
+  useEffect(() => {
+    if (getState()?.initialValues?.isUpdate) {
+      setIsUpdate(true);
+    } else {
+      (async () => {
+        const data = await fetchActivationKeys(10);
```
Comment (ldjebran) [inline src/components/form/ActivationKeysSelector.js:27]: Think the parameter is an object, and the limit is too small 

```suggestion
        const data = await fetchActivationKeys({limit=100});
```

**Review by ldjebran — COMMENTED** [review]
LGTM , left some comments

**Review by adeliaferreira — APPROVED** [review]
lgtm

**Pull request merged**